test/new-e2e/tests/windows/common/activedirectory/provisioner.go

```diff
@@ -0,0 +1,74 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package activedirectory
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/optional"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/fakeintake"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	provisionerBaseID = "aws-ec2vm-active-directory-"
+	defaultVMName     = "dcvm"
+)
+
+// Provisioner creates an Active Directory environment on a given VM.
+func Provisioner(opts ...ProvisionerOption) e2e.TypedProvisioner[Env] {
+	params := newProvisionerParams()
+	err := optional.ApplyOptions(params, opts)
+	if err != nil {
+		panic(fmt.Errorf("unable to apply ProvisionerOption, err: %w", err))
+	}
+
+	return e2e.NewTypedPulumiProvisioner[Env](provisionerBaseID+params.name, func(ctx *pulumi.Context, env *Env) error {
+		params := newProvisionerParams()
+		_ = optional.ApplyOptions(params, opts)
+
+		awsEnv, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+
+		env.Environment = awsEnv.CommonEnvironment
+
+		// JL: should the ec2 VM be customizable by the user?
```

Comment:
I think it'd be helpful if we could customize the Windows Server version here (2016, 2019, etc), so that if we needed to spin up a manual test environment for a specific issue on a specific server version. Probably doesn't _need_ to be in this PR, but if it isn't too much trouble, I could see it being useful.

Reply:
Understood - yeah, depending on complexity, it'd be nice to add it here or as a fast-follow PR 🤷